homeassistant/components/huawei_lte/switch.py

```diff
@@ -113,3 +120,41 @@ def _turn(self, state: bool) -> None:
     def icon(self) -> str:
         """Return switch icon."""
         return "mdi:signal" if self.is_on else "mdi:signal-off"
+
+
+@dataclass
+class HuaweiLteWifiGuestNetworkSwitch(HuaweiLteBaseSwitch):
+    """Huawei LTE WiFi guest network switch device."""
+
+    def __post_init__(self) -> None:
+        """Initialize identifiers."""
+        self.key = KEY_WLAN_WIFI_GUEST_NETWORK_SWITCH
+        self.item = "WifiEnable"
+
+    @property
+    def _entity_name(self) -> str:
+        return "WiFi guest network"
+
+    @property
+    def _device_unique_id(self) -> str:
+        return f"{self.key}.{self.item}"
+
+    @property
+    def is_on(self) -> bool:
+        """Return whether the switch is on."""
+        return self._raw_state == "1"
+
+    def _turn(self, state: bool) -> None:
+        self.router.client.wlan.wifi_guest_network_switch(state)
+        self._raw_state = "1" if state else "0"
+        self.schedule_update_ha_state()
+
+    @property
+    def icon(self) -> str:
+        """Return switch icon."""
+        return "mdi:wifi" if self.is_on else "mdi:wifi-off"
+
+    @property
+    def extra_state_attributes(self) -> dict[str, str]:
+        """Return the state attributes."""
+        return {"ssid": self.router.data[self.key].get("WifiSsid")}
```

Comment:
The rerurm value typing is missing optional.

Reply:
If the measurement data is relevant without knowing the main dtate of the switch the state attribute should be a separate sensor entity.